plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,845 @@
+                mso.fail_json(msg="Interface type cannot be changed.")
+
+            if domains:
+                # the existing domain list is updated by the new domain list
+                domain_uuid = []
+                # get names and uuids of existing domains
+                existing_physical_domains = {domain["name"]: domain["uuid"] for domain in template_info.get("domains", [])}
+                existing_l3_domains = {domain["name"]: domain["uuid"] for domain in template_info.get("l3Domains", [])}
+                for item in domains:
+                    if item in existing_physical_domains:
+                        domain_uuid.append(existing_physical_domains[item])
+                    elif item in existing_l3_domains:
+                        domain_uuid.append(existing_l3_domains[item])
+                    else:
+                        mso.fail_json(msg="Domain '{0}' not found in the template '{1}'.".format(item, template))
+                if set(domain_uuid) != set(match.details.get("domains")):
+                    ops.append(dict(op="replace", path="{0}/{1}/domains".format(path, match.index), value=domain_uuid))
+                match.details["domains"] = domain_uuid
+            elif domains == []:
+                # remove domains from the interface policy group if the list is empty
+                ops.append(dict(op="remove", path="{0}/{1}/domains".format(path, match.index)))
+                match.details.pop("domains", None)
+
+            if sync_e:
+                # existing_sync_e contains names and uuids of existing SyncE policies
+                existing_sync_e = {sync_e["name"]: sync_e["uuid"] for sync_e in template_info.get("syncEthIntfPolicies", [])}
+                if sync_e not in existing_sync_e:
+                    mso.fail_json(msg="SyncE policy '{0}' not found in the template '{1}'.".format(sync_e, template))
+                else:
+                    if existing_sync_e[sync_e] != match.details.get("syncEthPolicy"):
+                        ops.append(dict(op="replace", path="{0}/{1}/syncEthPolicy".format(path, match.index), value=existing_sync_e[sync_e]))
+                        match.details["syncEthPolicy"] = existing_sync_e[sync_e]
+
+            if access_macsec_policy:
+                if mcp and mcp["admin_state"] and (mcp["admin_state"] == "disabled" or match.details.get("mcp", {}).get("adminState") == "disabled"):
+                    mso.fail_json(msg="Access MACsec policy can only be set when MCP admin state is enabled.")
+                else:
+                    # existing_access_macsec_policy contains names and uuids of existing MACsec policies
+                    existing_access_macsec_policy = {macsec_policy["name"]: macsec_policy["uuid"] for macsec_policy in template_info.get("macsecPolicies", [])}
+                    if access_macsec_policy in existing_access_macsec_policy and (existing_access_macsec_policy[access_macsec_policy] != access_macsec_policy):
```

Comment:
what exactly is this condition `and (existing_access_macsec_policy[access_macsec_policy] != access_macsec_policy)`? aren't we comparing the name of a policy with the uuid of a policy? if so, this condition will always be true

Reply:
resolved